docs/topics/lombok.md

```diff
@@ -0,0 +1,149 @@
+
+> Generation of `@Builder` annotation won't be supported in the current prototype. However, you can give your feedback
+> about the plugin and vote for [`@Builder` in YouTrack](https://youtrack.jetbrains.com/issue/KT-46959).
+>
+{type="warning"}
+
+> Kotlin compiler ignores Lombok annotations if you use them in Kotlin code.
+>
+{type="note"}
+
+Please read below about how to apply the Lombok Kotlin compiler plugin to your Gradle/Maven build.
+
+## Gradle
+
+### Main settings
+
+Apply the `kotlin-plugin-lombok` Gradle plugin:
+
+```groovy
+plugins {
+    id "org.jetbrains.kotlin.plugin.lombok" version "%kotlinVersion%"
+}
+```
+
+Alternatively, you can use the `apply plugin` syntax:
+
+```groovy
+apply plugin: 'kotlin-plugin-lombok'
+```
+
+The Lombok itself should be set up as the Java annotation processor (not via kapt) or as the plugin as below:
+
+```groovy
+plugins {
+    id "io.freefair.lombok" version "5.3.0"
+}
+```
+Learn the [test project with examples of Lombok compiler plugin usage](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/nokapt).
+
+### Defining the place of lombok.config
+
+If you use the file `lombok.config` for [configuring Lombok](https://projectlombok.org/features/configuration),
+Kotlin Lombok compiler plugin should know where this file situates. Use the following syntax in your `build.gradle` file:
+
+```groovy
+kotlinLombok {
+    lombokConfigurationFile(file("lombok.config"))
+}
+```
+
+Learn the [test project with examples of Lombok compiler plugin and `lombok.config` usages](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_gradle/withconfig).
+
+## Maven
+
+Here's how to use Lombok Kotlin compiler plugin with Maven:
+
+```xml
+<plugin>
+    <groupId>org.jetbrains.kotlin</groupId>
+    <artifactId>kotlin-maven-plugin</artifactId>
+    <version>${kotlin.version}</version>
+    <configuration>
+        <compilerPlugins>
+            <plugin>lombok</plugin>
+        </compilerPlugins>
+        <pluginOptions>
+            <option>lombok:config=${project.basedir}/lombok.config</option>
+        </pluginOptions>
+    </configuration>
+    <dependencies>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-maven-lombok</artifactId>
+            <version>${kotlin.version}</version>
+        </dependency>
+    </dependencies>
+</plugin>
+```
+
+Please refer to the [Gradle](#gradle) section for detailed information about the Lombok Kotlin compiler plugin configuration.
+
+Learn the [test project example of Lombok compiler plugin and `lombok.config` usages](https://github.com/kotlin-hands-on/kotlin-lombok-examples/tree/master/kotlin_lombok_maven/nokapt).
+
+## Using the plugin along with kapt
+
+If you use [kapt](kapt.md) compiler plugin, note that Java's Lombok should be launched via annotation processor of Java's compiler.
```

Comment:
Couldn't we just say "Java Lombok"?

Reply:
Is it ok to say "If you use the kapt compiler plugin, you should launch Java Lombok via Java compiler annotation processor"?

Reply:
"note that Java's Lombok should be launched..." - just "launch Java Lombok..."
"should" is also a no-no: https://helpserver.labs.jb.gg/help/stardust/1.0/style-guide.html#modal-verbs

Reply:
>Couldn't we just say "Java Lombok"?

I'm not sure, let's wait for a proofread.

Reply:
> Java's Lombok should be launched by
This can be read in multiple meanings, this is why we're avoiding should and passive.
The exact instruction for a reader can be something like "Use a Java AP to launch Lombok".

Reply:
Does it work?

>By default, [kapt](kapt.md) compiler plugin launches all annotation processors and disables javac's annotation processors.
You need to change this behavior: enable Java's Lombok.

>If you use Gradle, add the option to the `build.gradle.kts` file to make javac's annotations
processors working: